frontend/src/components/problem/editor/OptionsPanel.tsx

```diff
@@ -0,0 +1,296 @@
+  const problemEditable = useProblemEditable(firebaseUser, newProblem);
+
+  // Handle updating of normal text fields
+  const handleChange = (e: any) => {
+    const { name, value } = e.target;
+    setNewProblem({
+      ...newProblem,
+      [name]: value,
+    });
+  };
+
+  // Handle updating of enum-type fields
+  const handleEnumChange = (name: string, value: any) => handleChange({ target: { name, value } });
+
+  // Handle verified change
+  const handleVerifiedChange = (value: boolean) => handleChange({ target: { name: 'verified', value } });
+
+  // Handle updating of problem inputs
+  const handleInputChange = (index: number, name: string, type: ProblemIOType) => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: newProblem.problemInputs.map((input, i) => {
+        if (index === i) {
+          return { name, type };
+        }
+        return input;
+      }),
+    });
+  };
+
+  const addTag = (newTag: ProblemTag) => {
+    setNewProblem({
+      ...newProblem,
+      problemTags: [...newProblem.problemTags, newTag],
+    });
+  };
+
+  const removeTag = (index: number) => {
+    setNewProblem({
+      ...newProblem,
+      problemTags: newProblem.problemTags.filter((_, i) => index !== i),
+    });
+  };
+
+  // Handle adding a new problem input for this problem
+  const addProblemInput = () => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: [...newProblem.problemInputs, { name: 'name', type: ProblemIOType.Integer }],
+    });
+  };
+
+  // Handle deleting a problem input for this problem
+  const deleteProblemInput = (index: number) => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: newProblem.problemInputs.filter((_, i) => index !== i),
+    });
+  };
+
+  const deleteProblemFunc = () => {
+    // eslint-disable-next-line no-alert
+    if (!window.confirm('Are you sure you want to delete this problem?')) {
+      return;
+    }
+
+    setLoading(true);
+    setError('');
+
+    deleteProblem(newProblem.problemId, token || '')
+      .then(() => {
+        setLoading(false);
+        history.replace('/');
+      })
+      .catch((err) => {
+        setError(err.message);
+        setLoading(false);
+      });
+  };
+
+  return (
+    <SidebarContent>
+      <HoverTooltip
+        visible={hoverVisible}
+        x={mousePosition.x}
+        y={mousePosition.y}
+      >
+        This variable name is likely invalid
+      </HoverTooltip>
+
+      <SmallHeaderText>Options</SmallHeaderText>
+      <SettingsContainer>
+        <VerifiedContainer
+          show={editMode && account?.role === AccountRole.Admin}
+        >
+          <ToggleButton
+            onChangeFunction={() => handleVerifiedChange(!newProblem.verified)}
+            editable={problemEditable}
+            checked={newProblem.verified}
+          />
+          <VerifiedText>
+            {newProblem.verified ? 'Verified' : 'Not Verified'}
+          </VerifiedText>
+        </VerifiedContainer>
```

Comment:
Thanks for finding this - the issue I think is that account info is only fetched manually, so if you refresh then the account will be null and it can't verify you as an admin account. I fixed this by adding a fetchAccount call on page load, after verifying the firebaseUser.